controllers/results/operation_results_test.go

```diff
@@ -0,0 +1,100 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package results
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Operation Results", func() {
+	var result *OperationResult
+
+	BeforeEach(func() {
+		result = &OperationResult{}
+		Expect(reflect.TypeOf(result)).To(Equal(reflect.TypeOf(&OperationResult{})))
+	})
+
+	It("returns an tuple instructing the reconcile loop to continue processing operations", func() {
+		Expect(ContinueProcessing()).Should(And(
+			HaveField("RequeueDelay", time.Duration(0)),
+			HaveField("RequeueRequest", false),
+			HaveField("CancelRequest", false),
+		))
+	})
+
+	It("// Requeue returns an (OperationResult, error) tuple instructing the reconcile loop to requeue the object", func() {
+		Expect(Requeue()).Should(And(
+			HaveField("RequeueDelay", time.Duration(0)),
+			HaveField("RequeueRequest", true),
+			HaveField("CancelRequest", false),
+		))
+	})
+
+	It("returns an tuple instructing the reconcile loop to requeue the object with the given reconcile error", func() {
+		result, err := RequeueWithError(errors.New("Test"))
+		Expect(fmt.Sprint(err)).To(Equal("Test"))
+		Expect(result).Should(And(
+			HaveField("RequeueDelay", time.Duration(0)),
+			HaveField("RequeueRequest", true),
+			HaveField("CancelRequest", false),
+		))
+	})
+
+	It("returns an tuple instructing the reconcile loop to requeue the object after the given delay", func() {
+		result, err := RequeueAfter(time.Duration(10), errors.New("Test"))
+		Expect(fmt.Sprint(err)).To(Equal("Test"))
+		Expect(result).Should(And(
+			HaveField("RequeueDelay", time.Duration(10)),
+			HaveField("RequeueRequest", true),
+			HaveField("CancelRequest", false),
+		))
+	})
+
+	It("returns an tuple instructing the reconcile loop to requeue the object in case of an error or to continue processing operations", func() {
+		result, err := RequeueOnErrorOrContinue(errors.New("Test"))
+		Expect(fmt.Sprint(err)).To(Equal("Test"))
+		Expect(result).Should(And(
+			HaveField("RequeueDelay", time.Duration(0)),
+			HaveField("RequeueRequest", false),
+			HaveField("CancelRequest", false),
+		))
+	})
+
+	It("returns an tuple instructing the reconcile loop to requeue the object in case of an error or to stop processing operations", func() {
+		result, err := RequeueOnErrorOrStop(errors.New("Test"))
+		Expect(fmt.Sprint(err)).To(Equal("Test"))
+		Expect(result).Should(And(
+			HaveField("RequeueDelay", time.Duration(0)),
+			HaveField("RequeueRequest", false),
+			HaveField("CancelRequest", true),
+		))
+	})
+
+	It("returns an tuple instructing the reconcile loop to stop processing operations", func() {
```

Comment:
`s/an tuple/a tuple/`

Reply:
I am not that bad in grammar. Just did not check what I was copying :joy:

Reply:
fixed.